### PR TITLE
refactor(ui-components): Set up ui-components library with ProgressEllipsis

### PR DESCRIPTION
**Pull request by carolinemodic**
Sets up a new shared library for ui components, with a basic initial component of ProgressEllipsis. I chose this one as it doesn't import any other components and has not diverged across our different repos. 

@eventualbuddha I'm not entirely sure I set up the linting properly... I couldn't get stylelint:fix to work properly, but was having issues with it in other repos as well so I just removed it. The rest of the linting setup is based on election-manager.

### CODE REVIEW

File: libs/ui-components/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/lib
```
Comment (beausmith) [inline libs/ui-components/.gitignore:1]: Ignore the `/coverage` dir:

```suggestion
/lib

# testing
/coverage
```

**Comment by beausmith** [comment]
Request: change from `ui-components` to `ui` so that the name is more general to the theme of user interface which could include other UI-related files: images, fonts, themes, etc.

**Review by beausmith — APPROVED** [review]
Sorry, I had intended to approve PR with the requested changes.

**Comment by carolinemodic** [comment]
Renamed to /ui
Fixed missing .gitignore things
Fixed tsconfigs in the importing apps not including @votingworks/ui, once I did this making changes to a component in /ui automatically triggers a re-compiling while running an app (i.e. election-manager) and the changes autoload.

File: apps/bmd/package.json [hunk 2]
```diff
@@ -89,6 +89,7 @@
     "@types/react-router-dom": "^5.1.5",
     "@types/styled-components": "^5.0.1",
     "@votingworks/ballot-encoder": "workspace:*",
+    "@votingworks/ui": "workspace:*",
```
Comment (eventualbuddha) [inline apps/bmd/package.json:92]: This should sort after `@votingworks/qrcode.react`, right?

File: libs/ui/.gitignore [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+# production
+/dist
+
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.enc.production.local
```
Comment (eventualbuddha) [inline libs/ui/.gitignore:19]:
```suggestion
.env.production.local
```

File: libs/ui/package.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
+{
+  "name": "@votingworks/ui",
+  "version": "1.0.0",
+  "private": true,
+  "description": "Generic UI components for VotingWorks UIs",
+  "license": "GPL-3.0",
+  "author": "Caroline Modic",
```
Comment (eventualbuddha) [inline libs/ui/package.json:7]: I probably have one with my name on it, but I intend to change any to this:
```suggestion
  "author": "VotingWorks <eng@voting.works>",
```

File: libs/ui/package.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+      "eslint --quiet --fix"
+    ],
+    "package.json": [
+      "sort-package-json"
+    ]
+  },
+  "dependencies": {
+    "@testing-library/react": "^11.2.6",
+    "@types/styled-components": "^5.1.9",
+    "prettier": "^2.1.1",
```
Comment (eventualbuddha) [inline libs/ui/package.json:40]: These belong in `devDependencies`, right?

**Review by eventualbuddha — APPROVED** [review]

**Pull request merged**